packages/discovery-provider/src/tasks/aggregates/__init__.py

```diff
@@ -47,23 +53,35 @@ def init_task_and_acquire_lock(
         # Attempt to acquire lock - do not block if unable to acquire
         have_lock = update_lock.acquire(blocking=False)
         if have_lock:
```

Comment:
nit: it looks like you removed the lock in every other task but left it here? also double-checking the reason it's safe to remove the locks is because there won't be any duplicate celery tasks in the queue?

Reply:
I did, it may be worth keeping those to at least block from running the same task twice at once. Ill definitely test on a stage node and see if it builds up

Reply:
This lock never gets released, did you want to undelete that part down below?